File: nan21.dnet.module.md/nan21.dnet.module.md.presenter/src/main/java/net/nan21/dnet/module/md/org/ds/model/StockLocatorDs.java
/* 
 * DNet eBusiness Suite
 * Copyright: Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */
package net.nan21.dnet.module.md.org.ds.model;

import net.nan21.dnet.core.api.annotation.Ds;
import net.nan21.dnet.core.api.annotation.DsField;
import net.nan21.dnet.core.api.annotation.Param;
import net.nan21.dnet.core.api.annotation.RefLookup;
import net.nan21.dnet.core.api.annotation.RefLookups;
import net.nan21.dnet.core.api.annotation.SortField;
import net.nan21.dnet.core.presenter.model.base.AbstractTypeDs;
import net.nan21.dnet.module.md.org.domain.entity.StockLocator;
import net.nan21.dnet.module.md.org.domain.entity.SubInventory;

@Ds(entity = StockLocator.class, sort = {@SortField(field = StockLocatorDs.f_name)})
@RefLookups({@RefLookup(refId = StockLocatorDs.f_subInventoryId, namedQuery = SubInventory.NQ_FIND_BY_NAME, params = {@Param(name = "pName", field = StockLocatorDs.f_subInventory)})})
public class StockLocatorDs extends AbstractTypeDs<StockLocator> {

	public static final String f_subInventoryId = "subInventoryId";
	public static final String f_subInventory = "subInventory";
	public static final String f_locatorTypeId = "locatorTypeId";
	public static final String f_locatorType = "locatorType";

	@DsField(join = "left", path = "subInventory.id")
	private Long subInventoryId;

	@DsField(join = "left", path = "subInventory.name")
	private String subInventory;

	@DsField(join = "left", path = "locatorType.id")
	private Long locatorTypeId;

	@DsField(join = "left", path = "locatorType.name")
	private String locatorType;

	public StockLocatorDs() {
		super();
	}

	public StockLocatorDs(StockLocator e) {
		super(e);
	}

	public Long getSubInventoryId() {
		return this.subInventoryId;
	}

	public void setSubInventoryId(Long subInventoryId) {
		this.subInventoryId = subInventoryId;
	}

	public String getSubInventory() {
		return this.subInventory;
	}

	public void setSubInventory(String subInventory) {
		this.subInventory = subInventory;
	}

	public Long getLocatorTypeId() {
		return this.locatorTypeId;
	}

	public void setLocatorTypeId(Long locatorTypeId) {
		this.locatorTypeId = locatorTypeId;
	}

	public String getLocatorType() {
		return this.locatorType;
	}

	public void setLocatorType(String locatorType) {
		this.locatorType = locatorType;
	}
}

File: nan21.dnet.module.md/nan21.dnet.module.md.ui.extjs/src/main/webapp/src/ds/StockLocatorDs.js
/*
 * DNet eBusiness Suite
 * Copyright: 2008-2012 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */
Ext.define("net.nan21.dnet.module.md.org.ds.model.StockLocatorDs" ,{
	extend: 'Ext.data.Model',
	fields: [
		{name:"active", type:"boolean"},
		{name:"clientId", type:"int", useNull:true},
		{name:"createdAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdBy", type:"string"},
		{name:"description", type:"string"},
		{name:"entityFQN", type:"string"},
		{name:"id", type:"int", useNull:true},
		{name:"locatorType", type:"string"},
		{name:"locatorTypeId", type:"int", useNull:true},
		{name:"modifiedAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedBy", type:"string"},
		{name:"name", type:"string"},
		{name:"subInventory", type:"string"},
		{name:"subInventoryId", type:"int", useNull:true},
		{name:"uuid", type:"string"},
		{name:"version", type:"int", useNull:true}
	],
	validations: [
		{field: "name", type: 'presence'}
	]
});
Ext.define("net.nan21.dnet.module.md.org.ds.model.StockLocatorDsFilter" ,{
	extend: 'Ext.data.Model',
	fields: [
		{name:"active", type:"boolean", useNull:true},
		{name:"createdAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdAt_From",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdAt_To",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdBy", type:"string"},
		{name:"description", type:"string"},
		{name:"entityFQN", type:"string"},
		{name:"id", type:"int", useNull:true},
		{name:"locatorType", type:"string"},
		{name:"locatorTypeId", type:"int", useNull:true},
		{name:"locatorTypeId_From",type:"int", useNull:true},
		{name:"locatorTypeId_To",type:"int", useNull:true},
		{name:"modifiedAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt_From",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt_To",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedBy", type:"string"},
		{name:"name", type:"string"},
		{name:"subInventory", type:"string"},
		{name:"subInventoryId", type:"int", useNull:true},
		{name:"subInventoryId_From",type:"int", useNull:true},
		{name:"subInventoryId_To",type:"int", useNull:true},
		{name:"uuid", type:"string"}
	]
});
Ext.define("net.nan21.dnet.module.md.org.ds.param.StockLocatorDsParam", {
	extend: 'Ext.data.Model',
	fields: [ 
	]
});
